### PR TITLE
Buffs painkillers to actually act like painkillers

### LINKED ISSUE #10984: Hideo Kojima's Metal Gear Solid V: The Phantom Pain
Missing limbs cause pain. Even if the limb was cleanly removed via surgery.
Having a couple missing limbs means permanent stacking pain stun unless you are forever hopped up on painkillers.
There's been a lot of confused players wondering why having no legs turned them into a fucking vegetable that can't even use their perfectly good hands. In part this also makes wheelchairs kind of pointless too.

### LINKED ISSUE #12871: Phantom Pain (Won't go away.)
(WEB REPORT BY: consedfu REMOTE: 172.93.109.202:7777)
>Revision a3fa295bada07af89fbf42932fcad66ea0d23ffb

>General description of the issue
For whatever reason the loss of a limb results in the inability to do anything at all due to immense pain.

>What you expected to happen
The pain to go away after the surgery ends.

>What actually happened
The pain didn't go away.

>Steps to reproduce if possible
Use circular saw to amputate any limb at all that won't result in immediate death.

### PR DESCRIPTION
**Pull request by 9600bauds**
Rather than a full rework of pain this is just a quick and dirty painkiller buff.

HOW PAIN WORKED BEFORE:
You have a "shock" variable and a "pain" variable.
Shock is the thing that makes you slow as fuck and eventually leaves you paralyzed on the floor (this is what people normally perceive as pain).
Pain is completely asymptomatic. Pain only ever matters if it goes over your pain resistance threshold: when over your pain resistance your shock starts slowly building up, otherwise your shock slowly decreases.
Painkillers only increase your pain resistance threshold (not by all that much). NOTHING EXCEPT TIME DECREASES YOUR SHOCK. So if you got a booboo and you come to medbay almost paralyzed from shock, you're already pretty much fucked: Even if you get enough painkillers to revert your pain, you still need to lie on the floor for like five fucking minutes, because that's the only thing that decreases your shock. Also, painkillers don't really go that far, so the doctors need to fix most of your damage first before the painkillers can even start reverting your pain (at which point, why give the guy painkillers? Just finish fixing his damage! Especially since painkillers are even less accessible than regular healing chemicals)
All in all: Pain is a BITCH that you can't really deal with at all, and painkillers are so worthless you may as well not make them.

CHANGES TO PAIN:
"Shock" and "Pain" still work the same. Missing limbs (the most painful thing in the game) have been dialed back a little.
Once your pain goes below your pain threshold, Tramadol also starts clearing your shock twice as fast, so there is actually a point to giving a guy painkillers after you fix his booboo.
Oxycodone now makes you completely numb! You feel no pain at all, which means nothing in your body will hurt... which means you cannot see your health at all. Can you believe this was actually there, half-implemented? Neat!
Numbness also lets you have surgery while awake.
Gave painkillers a lower metabolism because damn that shit goes through you like water through a colander when you're trying to ignore your broken arm (and especially Oxycodone since it requires plasma to make). Also, Oxycodone now results in 3 units instead of 1.
Missing limbs that weren't properly amputated give you a TON of pain, but this wasn't really clarified anywhere. When you have such a painful stump now you will get an actual pain message telling you what hurts.
Effects of pain not changed at all

OxycoCaptain walks over glass shards to see if it hurts

Fixes #12871
Fixes #10984
:cl: 
 * tweak: Tramadol will now clear shock from pain faster. Oxycodone now makes you completely immune to pain at the cost of not being able to see your own health. Lowered metabolism rate of both Tramadol and Oxycodone.
 * tweak: Improperly amputated limbs now hurt a little less and now give you an actual message telling you that they hurt.
 * bugfix: Fixed a bug where cutting off your arm/leg would cause your hand/foot to be perpetually set to "improperly amputated", thus doubling the pain it should cause and giving you sticky phantom pain even if you got your arm nub fixed.

### CODE REVIEW

File: code/modules/mob/living/carbon/shock.dm [hunk 1]
```diff
@@ -1,32 +1,26 @@
-/mob/living/var/traumatic_shock = 0
-/mob/living/carbon/var/shock_stage = 0
+/mob/living/carbon/var/pain_shock_stage = 0
+/mob/living/carbon/var/pain_level = 0 //Gets overwritten every tick! If you want to add pain "resistance" or constain pain, see update_pain_level()
+/mob/living/carbon/var/pain_numb = 0 // Ehen this is set, the mob isn't affected by shock or pain, but can't see their health.
```
Comment (Probe1) [inline code/modules/mob/living/carbon/shock.dm]: Ehen, whens evil brother.

**Comment by ghost** [comment]
https://youtu.be/_FrOQC-zEog

File: code/modules/organs/pain.dm [hunk 2]
```diff
@@ -75,27 +63,25 @@ mob/living/carbon/human/proc/custom_pain(var/message, var/flash_strength)
 	next_pain_time = world.time + 100
 
 mob/living/carbon/human/proc/handle_pain()
-	// not when sleeping
-
-	if(!feels_pain())
-		return
-
-	if(stat >= 2)
+	if(sleeping || stat >= DEAD)
```
Comment (Probe1) [inline code/modules/organs/pain.dm]: isn't DEAD the highest stat goes (2)?

mega death?

Comment (9600bauds) [reply code/modules/organs/pain.dm]: firefox_2017-01-02_04-09-52

**Comment by ScrambledSteak** [comment]
https://youtu.be/kXYiU_JCYtU

Official PR Theme

**Review by Probe1 — APPROVED** [review]
Nothing retarded jumped out at me.

File: code/modules/organs/pain.dm [hunk 3]
```diff
@@ -75,27 +63,25 @@ mob/living/carbon/human/proc/custom_pain(var/message, var/flash_strength)
...
-		// amputated limbs don't cause pain
-		if(!E.is_existing() || E.amputated)
-			continue
 		if(E.status & ORGAN_DEAD || E.status & ORGAN_PEG)
 			continue
+		// CLEANLY amputated limbs don't cause pain
+		else if(E.amputated)
+			continue
+		// your shredded nub of a fuckin' arm DOES, though
+		else if (!E.is_existing())
```
Comment (9600bauds) [inline code/modules/organs/pain.dm]: Reminder to add && parent.is_existing() here

**Comment by 9600bauds** [comment]
Ez xenoarch artifact effect: turns everyone comfortably numb for a while

**Comment by jakmak3** [comment]
OK while this is a excellent buff for tram and paracentomal oxycodone is still far to expensive to justify (it needs fucking plasma) Why not add to oxycodone that if it is in your system permits self surgery(with a higher failure rate ofcourse)

**Comment by sonime** [comment]
I think disregarding any pain effects is a strong enough buff as-is.

**Comment by jakmak3** [comment]
the problem is since it needs plasma not even the most powergamey of powermemers would think to use it in large enough amounts.

**Comment by BestName** [comment]
I think you're underestimating powergaymers.

**Comment by PJB3005** [comment]
Plasma is ridiculously easy to get what are you on about.

**Comment by icantthinkofanameritenow** [comment]
I think it's the part that it's rarely if ever given to chemistry. 

Anyway, wasn't oxycodone also supposed to be addictive too?

**Comment by Probe1** [comment]
It's given as often as it's asked for which is almost never.

**Comment by 9600bauds** [comment]
Made oxycodone result in 3 units instead of 1 (like the wiki said), I don't plan to remove the plasma requirement since chemistry is already too braindead simple and removing what little exclusivity it exists would be a crime. With oxycodone's lower metabolism you should have enough for at least 5 patients with just 5u plasma anyways.

**Comment by 9600bauds** [comment]
dreamseeker_2017-01-02_20-54-36
Does blurry text look too awful for the oxycodone high flavor messages?

**Comment by Exxion** [comment]
Yes, but do it anyway

Also oxycodone doesn't actually make you numb
But I guess Methylin also doesn't let monkeys use computers, so whatever

**Comment by 9600bauds** [comment]
It's SPACE OXYCODONE MADE WITH 100% REAL PLASMA

**Comment by Exxion** [comment]
It's always irked me a bit that a couple of our drugs are named after actual drugs, but that's a matter for another PR, if at all

**Comment by 9600bauds** [comment]
Also a little numb aura around your screen
spess space-hulk kappa 2017-01-02 210016
spess space-hulk kappa 2017-01-02 210022
As you can see, drugs open up your eyes to the world of nopower

**Comment by Probe1** [comment]
Needs more celtic.  Make this play https://www.youtube.com/watch?v=kXYiU_JCYtU

**Pull request merged**